### ISSUE TITLE
GraphQL Codegen does not understand relative plugins

### ISSUE THREAD

**Issue by gajus** [webpro/knip#349]
```
@graphql-codegen/./codegen/GraphQLResolverBuilderPlugin.cjs  codegen.yml
@graphql-codegen/./codegen/UserErrorsPlugin.cjs              codegen.yml
@graphql-codegen/[object Object]                             codegen.yml
```

For context, our config:

```yaml
generates:
  '../../packages/graphql-types/src/types.ts':
    config:
      enumsAsTypes: true
      immutableTypes: true
      scalars:
        SerializedMoney: 'SerializedMoney'
        SerializedUsd: 'SerializedMoney<''USD''>'
    documents:
      'src/documents/**/*.graphql':
        skipGraphQLImport: false
    plugins:
      - typescript
      - typescript-operations
      - typed-document-node
      - ./codegen/UserErrorsPlugin.cjs
      - add:
          content: "/* cSpell:disable */\nimport { type SerializedMoney } from '@contra/money';"
  'src/contra-api/__generated__/types.ts':
    config:
      avoidOptionals: true
      contextType: '../types.js#ResolverContext'
      enumsAsTypes: true
      immutableTypes: true
      mappers:
        ActivePaidProjectAnalytics: '../types.js#ActivePaidProjectAnalytics'
        AdminUserSearchResult: '../types.js#UserSearchResult'
        AnalyticsLocation: '../types.js#AnalyticsLocation'
        AnalyticsPortfolioProject: '../types.js#AnalyticsPortfolioProject'
        AnalyticsProductizedService: '../types.js#AnalyticsProductizedService'
        AnalyticsTrafficSource: '../types.js#AnalyticsTrafficSource'
        ApplicationAnalytics: '../types.js#ApplicationAnalytics'
        AppliedJobCandidate: '@contra/temporary-common/types.js#JobCandidate'
        AutonomousSystem: '../types.js#AutonomousSystem'
        AutoRatingDeductions: '../types.js#AutoRatingDeductions'
        AutoRatingIndividualProjectDeductions: '../types.js#AutoRatingIndividualProjectDeductions'
        AutoRatingIndividualServiceDeductions: '../types.js#AutoRatingIndividualServiceDeductions'
        BlogCategory: '../types.js#BlogCategory'
        BrandCustomization: '../types.js#BrandCustomization'
        CadenceFixedPricePaidProject: '../types.js#PaidProjectV2'
        CadenceFixedPricePaidProjectTerms: '@contra/data-accessors#PaidProjectTermTransform'
        ClientHomeActionItem: '../types.js#ClientHomeActionItemWithFilter'
        ClientOnboardingGoals: '../types.js#ClientOnboardingGoals'
        CollaborationRequest: '../types.js#CollaborationRequest'
        Collaborator: '../types.js#PortfolioProjectCollaborator'
        ConnectedAccounts: '../types.js#ConnectedAccounts'
        Contact: '@contra/data-accessors#ContactSelect'
        ContactExternalLink: '../types.js#ContactExternalLink'
        ContactInvite: '@contra/data-accessors#ContactInviteSelect'
        ContactNote: '../types.js#ContactNote'
        ContactSource: '@contra/data-accessors#ContactSourceSelect'
        ContactStage: '@contra/data-accessors#ContactStageSelect'
        ContactTag: '@contra/data-accessors#ContactTagSelect'
        ContactTaggable: '../types.js#ContactTaggable'
        ContraImage: '../types.js#ContraImage'
        ContraTransfer: '../types.js#MoneyTransaction'
        ContraVideo: '../types.js#ContraVideo'
        Conversation: '@contra/data-accessors#ConversationUserAccountSelect'
        Country: '../types.js#Country'
        CryptoCurrencyAddress: '../types.js#CryptoCurrencyAddress'
        CryptoPayout: '../types.js#MoneyTransaction'
        CryptoPayoutAccount: '../types.js#PayoutAccount'
        CustomDomain: '../types.js#CustomDomain'
        DiscoverPortfolioProjectsFacetValue: '../types.js#DiscoverSearchFacetValue'
        DiscoverProductizedServicesFacetValue: '../types.js#DiscoverSearchFacetValue'
        EmailSubscriptionSettings: '../types.js#EmailSubscriptionSettings'
        ExistingUserCollaborationRequest: '../types.js#CollaborationRequest'
        ExistingUserCollaborator: '../types.js#PortfolioProjectCollaborator'
        ExistingUserProductizedServiceCollaborationRequest: '../types.js#ProductizedServiceCollaborationRequest'
        ExistingUserProductizedServiceCollaborator: '../types.js#ProductizedServiceCollaborator'
        FeaturedUserProfileCategory: '../types.js#ProfessionalCategory'
        FixedPricePaidProject: '../types.js#PaidProjectV2'
        FixedPricePaidProjectTerms: '@contra/data-accessors#PaidProjectTermTransform'
        GoogleAccount: '../types.js#GoogleAccount'
        GooglePlaceLocation: '../types.js#GooglePlaceLocation'
        GuestInvoice: '@contra/temporary-common/domain/paidProject/services/invoices/GuestInvoiceService.js#GuestInvoice'
        HiringRequest: '../types.js#HiringRequest'
        HourlyRatePaidProject: '../types.js#PaidProjectV2'
        HourlyRatePaidProjectTerms: '@contra/data-accessors#PaidProjectTermTransform'
        IncomeAnalytics: '../types.js#IncomeAnalytics'
        IndependentPortfolioTemplate: '../types.js#IndependentPortfolioTemplate'
        IndependentPortfolioTemplateTheme: '../types.js#IndependentPortfolioTemplateTheme'
        IndustryCategory: '../types.js#IndustryCategory'
        IndyOnboardingGoals: '../types.js#IndyOnboardingGoals'
        InquiryAnalytics: '../types.js#InquiryAnalytics'
        IntroductionVideo: '../types.js#IntroductionVideo'
        InvitedJobCandidate: '@contra/temporary-common/types.js#JobCandidate'
        Invoice: '../types.js#Invoice'
        InvoiceAnalytics: '../types.js#InvoiceAnalytics'
        InvoiceFooterItem: '../types.js#InvoiceFooterItem'
        InvoiceLineItem: '../types.js#InvoiceLineItem'
        IpAddress: '../types.js#IpAddress'
        Job: '@contra/temporary-common/types.js#Job'
        JobCandidate: '@contra/temporary-common/types.js#JobCandidate'
        JobCandidateResponse: '@contra/temporary-common/types.js#JobCandidateResponse'
        JobPosting: '@contra/temporary-common/types.js#JobPosting'
        JobPostingApplication: '../types.js#JobPostingApplication'
        JobPostingCategory: '../types.js#JobPostingCategory'
        JobPostingDeliverable: '../types.js#JobPostingDeliverable'
        JobPostingMatch: '../types.js#JobPostingMatch'
        JobPostingMoreMatchesRequest: '@contra/data-accessors#JobPostingMoreMatchesRequestSelect'
        JobPostingSubcategory: '../types.js#JobPostingSubcategory'
        MilestonePaidProject: '../types.js#PaidProjectV2'
        MilestonePaidProjectTerms: '@contra/data-accessors#PaidProjectTermTransform'
        MoneyTransaction: '../types.js#MoneyTransaction'
        MoneyTransactionPaidProjectDetails: '../types.js#MoneyTransactionPaidProjectDetails'
        NewNotificationEvent: '../types.js#NewNotificationEvent'
        NewUserCollaborationRequest: '../types.js#CollaborationRequest'
        NewUserCollaborator: '../types.js#PortfolioProjectCollaborator'
        NewUserProductizedServiceCollaborationRequest: '../types.js#ProductizedServiceCollaborationRequest'
        NewUserProductizedServiceCollaborator: '../types.js#ProductizedServiceCollaborator'
        NotificationV2: '../types.js#NotificationV2'
        OngoingFixedPaidProjectCycle: '{ id: number; amount: USD; endDate: string; recurringPaidProjectConfigurationId: number; }'
        OngoingFixedRatePaidProject: '../types.js#PaidProjectV2'
        OngoingHourlyRatePaidProject: '../types.js#PaidProjectV2'
        OngoingHourlyRatePaidProjectTerms: '@contra/data-accessors#PaidProjectTermTransform'
        Organization: '@contra/temporary-common/domain/organization/repositories/OrganizationRepository.js#Organization'
        OrganizationMember: '{ createdAt: Date; id: number; updatedAt: Date; organizationId: number; userAccountId: number; role: "ADMIN" | "MEMBER" | "CONTRA_ADMIN"; emailAddress: string;}'
        OrganizationMemberInvite: '@contra/data-accessors#OrganizationMemberInviteSelect'
        OrganizationTeam: '@contra/data-accessors#OrganizationTeamSelect'
        Outcome: '../types.js#Outcome'
        PaidProject: '../types.js#PaidProject'
        PaidProjectDeliverable: '../types.js#PaidProjectDeliverable'
        PaidProjectFee: '../types.js#PaidProject'
        PaidProjectFixedPriceFee: '../types.js#PaidProject'
        PaidProjectInquiry: '../types.js#PaidProjectInquiry'
        PaidProjectInvite: '../types.js#PaidProjectInvite'
        PaidProjectMilestone: '@contra/temporary-common/types.js#PaidProjectMilestone'
        PaidProjectMilestoneV2: '@contra/temporary-common/types.js#PaidProjectMilestone'
        PaidProjectOngoingFixedTerms: '@contra/data-accessors#PaidProjectTermTransform'
        PaidProjectRateFee: '../types.js#PaidProject'
        PaidProjectSubmission: '@contra/data-accessors#PaidProjectSubmissionSelect'
        PaidProjectTerms: '@contra/data-accessors#PaidProjectTermTransform'
        PaidProjectTermsContract: '@contra/data-accessors#PaidProjectTermTransform'
        PaidProjectTermsContractSignature: '@contra/data-accessors#ContractSignatureSelect'
        PaidProjectTermsContractV2: '@contra/data-accessors#ContractSelect'
        PaidProjectV2: '../types.js#PaidProjectV2'
        PaidProjectV2Party: '../types.js#PaidProjectV2Party'
        PaymentMethodV2: '../types.js#ExtendedPaymentMethod'
        PayoutAccount: '../types.js#PayoutAccount'
        PaypalPayout: '../types.js#MoneyTransaction'
        PaypalPayoutAccount: '../types.js#PayoutAccount'
        PendingPaidProjectInquiry: '../types.js#PendingPaidProjectInquiry'
        PersonaIdentity: '../types.js#PersonaIdentity'
        PersonaInquiry: '../types.js#PersonaInquiry'
        PersonaVerificationStatusChangedEvent: '../types.js#PersonaVerificationStatusChangedEvent'
        PortfolioProject: '../types.js#PortfolioProject'
        PortfolioProjectExternalLink: '../types.js#PortfolioProjectExternalLink'
        PortfolioProjectList: '../types.js#PortfolioProjectList'
        PortfolioProjectTemplate: '../types.js#PortfolioProjectTemplate'
        PortfolioProjectViewAnalytics: '../types.js#PortfolioProjectViewAnalytics'
        PotentialCollaborator: '../types.js#PortfolioProjectCollaborator'
        PrivateContactMetadata: '../types.js#Contact'
        ProductizedService: '../types.js#ProductizedService'
        ProductizedServiceCollaborationRequest: '../types.js#ProductizedServiceCollaborationRequest'
        ProductizedServiceCollaborator: '../types.js#ProductizedServiceCollaborator'
        ProductizedServiceDeliverable: '../types.js#ProductizedServiceDeliverable'
        ProductizedServicePurchase: '../types.js#ProductizedServicePurchase'
        ProductizedServiceViewAnalytics: '../types.js#ProductizedServiceViewAnalytics'
        ProfessionalCategory: '../types.js#ProfessionalCategory'
        ProjectHistoryEvent: '@contra/temporary-common/types.js#ProjectHistoryEvent'
        ProposalAnalytics: '../types.js#ProposalAnalytics'
        PublicOrganizationMember: '{ createdAt: Date; id: number; updatedAt: Date; organizationId: number; userAccountId: number;}'
        RecommendationRequest: '../types.js#RecommendationRequest'
        ReferralEmail: '../types.js#ReferralEmail'
        Role: '../types.js#ProfessionalRole'
        ShortUrl: '../types.js#ShortUrl'
        SocialMediaSite: '../types.js#SocialMediaSite'
        StripeAccount: '../types.js#UserAccount'
        StripeExternalAccount: '../types.js#StripeExternalAccount'
        StripeIdentity: '../types.js#StripeIdentity'
        StripePaymentCardV2: '@contra/temporary-common/domain/payment/repositories/StripePaymentMethodRepository.js#PaymentMethodCard'
        StripePaymentV2: '../types.js#MoneyTransaction'
        StripePayoutAccount: '../types.js#PayoutAccount'
        StripePayoutV2: '../types.js#MoneyTransaction'
        StripeRefundV2: '../types.js#MoneyTransaction'
        StripeSepaDebitPaymentMethodV2: '@contra/temporary-common/domain/payment/repositories/StripePaymentMethodRepository.js#PaymentMethodSepaDebit'
        StripeTransferV2: '../types.js#MoneyTransaction'
        StripeUsBankAccountPaymentMethodV2: '@contra/temporary-common/domain/payment/repositories/StripePaymentMethodRepository.js#PaymentMethodUsBankAccount'
        StripeVerificationStatusChangedEvent: '../types.js#StripeVerificationStatusChangedEvent'
        Tag: '../types.js#Tag'
        Testimonial: '../types.js#Testimonial'
        ThemeBackgroundStyle: '../types.js#ThemeBackgroundStyle'
        ThemeCustomization: '../types.js#ThemeCustomization'
        ThemeFontPack: '../types.js#ThemeFontPack'
        ThemeProjectLayout: '../types.js#ThemeProjectLayout'
        ThemeServiceLayout: '../types.js#ThemeServiceLayout'
        TikTokAccount: '../types.js#TikTokAccount'
        Timezone: '../types.js#Timezone'
        Tool: '../types.js#ProfessionalTool'
        UploadedFile: '@contra/data-accessors#UploadedFileSelect'
        UserAccount: '../types.js#UserAccount'
        UserAccountClientHomeStats: '../types.js#UserAccount'
        UserAccountCreatedJobPostingsConnection: '{count: number; edges: Array<{id: number, cursor: string}>, pageInfo: PageInfo, userAccountId: number}'
        UserAccountIdentity: '../types.js#UserAccountIdentity'
        UserAccountJobPostingsConnection: '{count: number; edges: Array<{id: number, cursor: string}>, pageInfo: PageInfo, userAccountId: number}'
        UserAccountOnboardingProgress: '../types.js#OnboardingProgress'
        UserAccountOpportunitiesConnection: '{count: number; edges: Array<{id: number, cursor: string}>, pageInfo: PageInfo, userAccountId: number}'
        UserAccountPaidProjectConnection: '{count: number; edges: Array<{id: number, cursor: string}>, pageInfo: PageInfo, userAccountId: number}'
        UserList: '../types.js#UserList'
        UserProfile: '../types.js#UserProfile'
        UserProfileAbout: '../types.js#UserProfileAbout'
        UserProfileAnalytics: '../types.js#UserProfileAnalytics'
        UserProfileExternalLink: '../types.js#UserProfileExternalLink'
        UserProfileImpressionAnalytics: '../types.js#UserProfileImpressionAnalytics'
        UserProfileImpressionViewer: '../types.js#UserProfileImpressionViewer'
        UserProfileImpressionViewers: '../types.js#UserProfileImpressionViewers'
        UserProfileProductizedServiceEdge: "{ isCollaborative: boolean; sortOrder: number; userProfileId: number; visibilityStatus: 'HIDDEN_CANNOT_CHANGE' | 'HIDDEN' | 'VISIBLE'; }"
        UserProfileRating: '../types.js#UserProfileRating'
        UserProfileViewAnalytics: '../types.js#UserProfileViewAnalytics'
        UserProfileViewer: '../types.js#UserProfileViewer'
        UserProfileViewers: '../types.js#UserProfileViewers'
        UserReview: '../types.js#UserReview'
        UserSearchResult: '../types.js#UserSearchResult'
        VerifiedIdentity: '../types.js#VerifiedIdentity'
        WaitList: '../types.js#UserWaitList'
        WorkPreferences: '../types.js#WorkPreferences'
        Workspace: '@contra/temporary-common/domain/workspace/WorkspaceService.js#Workspace'
      mapperTypeSuffix: 'Model'
      maybeValue: 'T | null | undefined'
      resolverTypeWrapperSignature: Promise<DeepPartial<T>> | DeepPartial<T>
      scalars:
        Date: 'Date'
        DateTime: 'Date'
        JSON: '@contra/types#Json'
        JSONObject: '@contra/types#Json'
        MoneyValue: 'string'
        SafeInt: 'number'
        SafeURL: 'string'
        SerializedMoney: 'ContraMoney'
        SerializedUsd: 'USD'
      useTypeImports: true
    plugins:
      - typescript
      - typescript-resolvers
      - ./codegen/UserErrorsPlugin.cjs
      - ./codegen/GraphQLResolverBuilderPlugin.cjs
      - add:
          content: "/* cSpell:disable */\nimport type { DeepPartial } from 'utility-types';\nimport { type Money as ContraMoney, type USD } from '@contra/money';"
schema:
  - node_modules/@contra/graphql-schema/index.js

```

**Comment by webpro** [comment]
:rocket: _This issue has been resolved in v2.41.3. See [Release 2.41.3](https://github.com/webpro/knip/releases/tag/2.41.3) for release notes._

**Comment by webpro** [comment]
Thanks Gajus, should be better now :)

**Comment by gajus** [comment]
Thank you!

**Comment by webpro** [comment]
:rocket: _This issue has been resolved in v3.0.0-canary.4. See [Release 3.0.0-canary.4](https://github.com/webpro/knip/releases/tag/3.0.0-canary.4) for release notes._
